magda-web-client/src/Components/Dataset/Search/Facets/FacetTemporal.js

```diff
@@ -131,50 +136,90 @@ class FacetTemporal extends Component {
         ]);
     }
 
+    onFocus() {
+        this.setState({
+            prompt: ""
+        });
+    }
+
     selectStartYear(startYear) {
+        const startBeforeEnd = this.checkStartAndEndDate();
+        if (!startBeforeEnd) {
+            this.setPrompt("End date is earlier than start date.");
+        }
         this.setState({
             startYear
         });
     }
 
     selectEndYear(endYear) {
+        const startBeforeEnd = this.checkStartAndEndDate();
+        if (!startBeforeEnd) {
+            this.setPrompt("End date is earlier than start date.");
+        }
         this.setState({
             endYear
         });
     }
 
     selectStartMonth(startMonth) {
+        const startBeforeEnd = this.checkStartAndEndDate();
+        if (!startBeforeEnd) {
+            this.setPrompt("End date is earlier than start date.");
+        }
         this.setState({
             startMonth
         });
     }
 
     selectEndMonth(endMonth) {
+        const startBeforeEnd = this.checkStartAndEndDate();
+        if (!startBeforeEnd) {
```

Comment:
Why not just do this bit in `checkStartAndEndDate`?

Reply:
Because it might be good to keep this a purely "is the start date-end date combo valid?" function. I suppose we can add a wrapper to this function (to add the warning) to deal with the annoying code duplication.

Reply:
Done in 950b1d5